src/Equations/DreicerNeuralNetwork.cpp

```diff
@@ -82,13 +84,48 @@ real_t DreicerNeuralNetwork::RunawayRate(
     real_t logNfree = log(nfree);
     real_t free_tot = nfree / ntot;
     real_t logTheta = log(T/Constants::mc2inEV);
+    
+    return nn_fix(fabs(E)/ED, logTheta, Zeff, Zeff0, Z0Z, Z0_Z,
+            logNfree, free_tot, nfree, tauEE, true);
+}
 
-    real_t rr = RunawayRate_derived_params(
-        fabs(E)/ED, logTheta, Zeff, Zeff0, Z0Z, Z0_Z,
-        logNfree, free_tot
-    );
-
-    return 4.0/(3.0*M_SQRTPI)*(nfree/tauEE) * rr;
+/**
+*Extrapolating the result from the NN such that it can be used for small E/ED(The
+*NN has not been trained for those values). Parameters are explained in the 
+*RunawayRate_derived_params function.
+*use_fix :  If enabled the return value will be an extrapolation of the form
+*exp(a * x^b)
+*/
+real_t DreicerNeuralNetwork::nn_fix(
+    const real_t EDD, const real_t logTheta, const real_t Zeff,
+    const real_t Zeff0, const real_t Z0Z, const real_t Z0_Z, const real_t logNfree, 
+    const real_t nfree_ntot, const real_t nfree, const real_t tauEE, bool use_fix
+) {
+    real_t boundary = 0.015;
+    real_t dE = 0.0001;
+    if(use_fix == true & EDD < boundary){
+        real_t gamma = 4.0/(3.0*M_SQRTPI) * RunawayRate_derived_params(
+            boundary, logTheta, Zeff, Zeff0, Z0Z, Z0_Z,
+            logNfree, nfree_ntot);
+        real_t gamma_dE = 4.0/(3.0*M_SQRTPI) * RunawayRate_derived_params(
+            boundary+dE, logTheta, Zeff, Zeff0, Z0Z, Z0_Z,
+            logNfree, nfree_ntot);
+
+        real_t A = log(gamma);
+        real_t B = (log(gamma_dE) - log(gamma))/dE;
```

Comment:
I don't think this is really covered in numerics courses at Chalmers, but there are actually some neat tricks for evaluating numerical derivatives. The Numerical Recipes book (@IdaEkmark should have a copy) describes this in section 5.7. The conclusion though is basically that if you are constrained to using a forward difference, ``dE = sqrt(eps) * x_c``, where ``eps`` is the machine epsilon (~1e-16 in double precision floating-point) and ``x_c`` is the "characteristic scale" over which the function to differentiate varies. This can be difficult to estimate, so often one takes ``x_c = x``, which would be ``x_c = boundary`` in the notation used here.

If possible to use, central differences give better accuracy (but for the purposes here a forward difference is probably fine).